metaseq/checkpoint_utils.py

```diff
@@ -163,23 +163,44 @@ def _launch_sbatch_for_checkpoint_copy(
             nfs_dir += "/"
         copy_script_dir = os.path.join(nfs_dir, cp_script_dir)
         os.makedirs(copy_script_dir, exist_ok=True)
-        sbatch_run_file = os.path.join(
-            copy_script_dir, f"_cp_sbatch_script_{num_update}.sh"
+
+        all_slurm_nodes = os.environ["SLURM_NODELIST"]
+        slurm_node_list = (
+            subprocess.check_output(
+                f"scontrol show hostnames {all_slurm_nodes}", shell=True
+            )
+            .decode()
+            .strip()
+            .split("\n")
         )
-        slurm_nodes = os.environ["SLURM_NODELIST"]
-        with open(sbatch_run_file, "w") as f:
-            f.write(
-                SBATCH_CHECKPOINT_COPY_CMD.format(
-                    oss_dir=oss_dir,
-                    slurm_nodes=slurm_nodes,
-                    local_dir=cfg.save_dir,
-                    num_files=num_files_per_host,
-                    nfs_dir=nfs_upload_path[4:],
-                    cp_script_dir=cp_script_dir,
-                    num_update=num_update,
+        num_hosts_per_chunk = 8
+        for idx in range(
+            0, len(slurm_node_list), num_hosts_per_chunk
```

Comment:
Can we launch so many jobs if we use a lot of nodes for the training? Could there be an issue if not all are getting resources in time?

Reply:
These are smol - asking for 4 cpu cores each (probably could do with just 1...) and no GPUs. We schedule 16 of them for a 128 host job, and they all end up scheduled on the same node :)